server/src/main/java/org/elasticsearch/index/mapper/flattened/FlattenedFieldMapper.java

```diff
@@ -225,10 +253,19 @@ public static final class KeyedFlattenedFieldType extends StringFieldType {
             );
             this.key = key;
             this.rootName = rootName;
+            this.isDimension = isDimension;
         }
 
         private KeyedFlattenedFieldType(String rootName, String key, RootFlattenedFieldType ref) {
-            this(rootName, ref.isIndexed(), ref.hasDocValues(), key, ref.splitQueriesOnWhitespace, ref.meta());
+            this(
+                rootName,
+                ref.isIndexed(),
+                ref.hasDocValues(),
+                key,
+                ref.splitQueriesOnWhitespace,
+                ref.meta(),
+                ref.dimensions.contains(key)
```

Comment:
I wonder if folks will try and use glob patterns here.

Reply:
Certainly `*` feels useful.

Reply:
We asked for wildcard support...it is not required by integrations. We decided not to support it because it makes more difficult to track the number of dimensions. We have a limit on the number of dimensions and being able to easily check how many we have in the mappings might make things like debugging easier.

Reply:
Also, I think when it comes to dimensions the set of dimensions to use for a metric should be "well-known"...which means whoever if using this should know exactly which fields are dimensions and which not.